diff --git a/node_modules/@sentry/node/cjs/integrations/undici/index.js b/node_modules/@sentry/node/cjs/integrations/undici/index.js
index b0c1053..a6fb0ad 100644
--- a/node_modules/@sentry/node/cjs/integrations/undici/index.js
+++ b/node_modules/@sentry/node/cjs/integrations/undici/index.js
@@ -245,16 +245,21 @@ function setHeadersOnRequest(
   sentryTrace,
   sentryBaggageHeader,
 ) {
-  const headerLines = request.headers.split('\r\n');
-  const hasSentryHeaders = headerLines.some(headerLine => headerLine.startsWith('sentry-trace:'));
+  try {
+    const headerLines = request.headers.split('\r\n');
+    const hasSentryHeaders = headerLines.some(headerLine => headerLine.startsWith('sentry-trace:'));
     
-  if (hasSentryHeaders) {
-    return;
-  }
+    if (hasSentryHeaders) {
+      return;
+    }
     
-  request.addHeader('sentry-trace', sentryTrace);
-  if (sentryBaggageHeader) {
-    request.addHeader('baggage', sentryBaggageHeader);
+    request.addHeader('sentry-trace', sentryTrace);
+    if (sentryBaggageHeader) {
+      request.addHeader('baggage', sentryBaggageHeader);
+    }
+  } catch(e) {
+    console.log('====== Error undici ======', e)
+    console.log(request.headers)
   }
 }
 
